[main] source-build-lite pipeline is failing due to missing CrossgenLayout targets file

Failure in `CentOSStream9_Offline_PreviousSourceBuiltSdk_x64` leg.

Full log: https://dev.azure.com/dnceng/internal/_build/results?buildId=2646153&view=logs&j=2efcdee0-63a0-5f4c-6890-20d889251d12&t=687ca494-57d2-542a-c494-988bec6c615f

```
      Restored /__w/1/s/src/sdk/artifacts/obj/redist-installer/Release/TemplatePackageDownloader/TemplatePackageDownloader.csproj (in 229 ms).
    /__w/1/s/src/sdk/src/Installer/redist-installer/targets/GenerateLayout.targets(534,11): error MSB4057: The target "CrossgenLayout" does not exist in the project. [/__w/1/s/src/sdk/src/Installer/redist-installer/redist-installer.csproj]
##[error]src/sdk/src/Installer/redist-installer/targets/GenerateLayout.targets(534,11): error MSB4057: (NETCORE_ENGINEERING_TELEMETRY=Build) The target "CrossgenLayout" does not exist in the project.
```

@ViktorHofer this might be related to your recent changes in https://github.com/dotnet/sdk/commits/main/src/Installer?

--- a/src/Installer/redist-installer/Directory.Build.targets
+++ b/src/Installer/redist-installer/Directory.Build.targets
@@ -14,9 +14,7 @@
   <Import Project="targets\BundledManifests.targets" />
   <Import Project="targets\BundledDotnetTools.targets" />
   <Import Project="targets\GenerateBundledVersions.targets" />
-  <!-- Crossgen is currently not supported on the s390x, ppc64le architecture as using mono instead of CoreCLR. -->
-  <Import Project="targets\Crossgen.targets"
-          Condition="'$(DISABLE_CROSSGEN)' == '' AND '$(Architecture)' != 's390x' AND '$(Architecture)' != 'ppc64le'" />
+  <Import Project="targets\Crossgen.targets" />
   <Import Project="targets\GenerateLayout.targets" />
   <Import Project="targets\GenerateArchives.targets" Condition="'$(PackInstaller)' != 'false'"/>
   <Import Project="targets\GenerateMSIs.targets" />

--- a/src/Installer/redist-installer/targets/Crossgen.targets
+++ b/src/Installer/redist-installer/targets/Crossgen.targets
@@ -1,6 +1,9 @@
 <Project>
 
   <PropertyGroup>
+    <!-- Crossgen is currently not supported on the s390x, ppc64le architecture as using mono instead of CoreCLR. -->
+    <IsCrossgenSupported Condition="'$(DISABLE_CROSSGEN)' == '' AND '$(Architecture)' != 's390x' AND '$(Architecture)' != 'ppc64le'">true</IsCrossgenSupported>
+
     <Crossgen2Rid>$(HostOSName)-$(BuildArchitecture)</Crossgen2Rid>
     <Crossgen2Rid Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(SharedFrameworkRid)</Crossgen2Rid>
 
@@ -8,11 +11,11 @@
   </PropertyGroup>
 
   <!-- Download the runtime package with the crossgen executable in it -->
-  <ItemGroup>
+  <ItemGroup Condition="'$(IsCrossgenSupported)' == 'true'">
     <PackageDownload Include="$(RuntimeNETCrossgenPackageName)" Version="[$(MicrosoftNETCoreAppRuntimePackageVersion)]" />
   </ItemGroup>
 
-  <Target Name="CrossgenLayout">
+  <Target Name="CrossgenLayout" Condition="'$(IsCrossgenSupported)' == 'true'">
     <PropertyGroup>
       <CrossgenPath>$(NuGetPackageRoot)$(RuntimeNETCrossgenPackageName)/$(MicrosoftNETCoreAppRuntimePackageVersion)/tools/crossgen2$(ExeExtension)</CrossgenPath>
       <CreateCrossgenSymbols Condition="'$(CreateCrossgenSymbols)' == ''">true</CreateCrossgenSymbols>

